gdsfactory/add_ports.py

```diff
@@ -158,9 +160,12 @@ def add_ports_from_markers_center(
     port_name_prefix = port_name_prefix or port_name_prefix_default
 
     pin_layer = gf.get_layer(pin_layer)
-    layer = gf.get_layer(layer)
 
-    polygons = component.get_polygons()
+    polygons = component.get_polygons(by="index")
```

Comment:
Is there any reason why we still use the crowbar to determine whether a layer is empty?

Why not use something like `c.bbox(layer_index).empty()`?

This would be the equivalent of querying all measurements to determine whether device_x has a measurement.

Also, if you want the bounding box in um, query directly for the um bbox with `c.dbbox()`